apps/site/assets/ts/stop/components/amenities/FareSalesAmenityCard.tsx

```diff
@@ -1,13 +1,120 @@
 import React from "react";
-import AmenityCard from "./AmenityCard";
+import AmenityCard, { AmenityLink, AmenityModal } from "./AmenityCard";
 import { faresIcon } from "../../../helpers/icon";
+import { Stop } from "../../../__v3api";
+import useFetch from "../../../hooks/useFetch";
+import { FetchState, FetchStatus } from "../../../helpers/use-fetch";
+import Loading from "../../../components/Loading";
 
-const FareSalesAmenityCard = (): JSX.Element => {
-  const icon = (
-    <span className="m-stop-page__icon">{faresIcon("c-svg__icon")}</span>
+const FareRetailLocations = (stop: Stop): JSX.Element => (
+  <>
+    <h2 className="h3 mb-8">Fare Retail Locations</h2>
+    <p className="mt-0">
+      This stop does not have fare vending machines, but you can purchase fares
+      at a{" "}
+      <a
+        href={`/fares/retail-sales-locations?location[address]=${stop.address}&location[latitude]=${stop.latitude}&location[longitude]=${stop.longitude}`}
+      >
+        nearby retail location
+      </a>{" "}
+      or pay with cash on any bus.
+    </p>
+  </>
+);
+
+const FareVendingMachines = (): JSX.Element => (
+  <>
+    <h2 className="h3 mb-8">Fare Vending Machines</h2>
+    <img
+      className="img-fluid mt-1 mb-1"
+      src="https://cdn.mbta.com/sites/default/files/styles/max_2600x2600/public/media/2021-11/Fare-Transformation-new-fvms-Nov2021.jpg"
+      alt="MBTA fare vending machines"
+      style={{ maxHeight: "250px" }}
+    />
+    <p className="mt-0 mb-0">At fare vending machines, you can:</p>
+    <ul>
+      <li>Purchase a CharlieCard or reload an existing one</li>
+      <li>Purchase tappable CharlieTickets</li>
+    </ul>
+    <p className="mt-0 mb-0">You can pay for your fare with:</p>
+    <ul>
+      <li>Apple Pay</li>
+      <li>Google Pay</li>
+      <li>Samsung Pay</li>
+      <li>Credit/debit cards</li>
+      <li>EBT card</li>
+      <li>Cash</li>
+    </ul>
+  </>
+);
+
+const fareTable = (tableData: [string, string][]): React.ReactElement => (
+  <>
+    <h2 id="fare-types" className="h3 mt-8">
+      Fare Types
+    </h2>
+    <table aria-labelledby="fare-types" className="fare-sales-table">
+      <thead>
+        <tr>
+          <th scope="col">Trip Type</th>
+          <th scope="col">Cost</th>
+        </tr>
+      </thead>
+      <tbody>
+        {tableData.map(([name, price]) => (
+          <tr key={name}>
+            <td>{name}</td>
+            <td>{price}</td>
+          </tr>
+        ))}
+      </tbody>
+    </table>
+  </>
+);
+
+const renderFareTable = ({
+  status,
+  data
+}: FetchState<[string, string][]>): React.ReactElement | null => {
+  switch (status) {
+    case FetchStatus.Loading:
+      return <Loading />;
+    default:
+      if (data && data.length > 0) return fareTable(data);
+      return null;
```

Comment:
I feel like there should be a error state here instead of null, if we go this route

Reply:
As in there should be a state for success, and then the default is the error state

Reply:
After consulting with Anna on this, we've decided that hiding the table and heading on error would be best.
So, this `default: return null` is in fact the default error state. :grin: